### ISSUE TITLE
StarCoder-trained VLLM vs. HF: Why the output is different?

### ISSUE THREAD

**Issue by matrixssy** [vllm-project/vllm#1979]
Hi,

Thank you very much for your contributions to inference acceleration. I am currently having a problem. I am using a model trained by StarCoder, and I am using the same input ids for inference. However, I found that the output of vllm is very different from the output of HF. vllm seems to be constantly outputting ```\n```. How can I solve this problem?

Thank you for your help.

**Comment by matrixssy** [comment]
from vllm import LLM, SamplingParams
from transformers import AutoTokenizer, AutoModelForCausalLM

path = 'models/XXX'
prompt = 'xxx'

# HF ways
model = AutoModelForCausalLM.from_pretrained(path, torch_dtype=torch.float16, device_map="auto")
tokenizer = AutoTokenizer.from_pretrained(path, fast_tokenizer=True, padding_size="left")
inputs = tokenizer(prompt, return_tensors="pt", add_special_tokens=True)
output = model.generate(inputs.input_ids.cuda(), max_new_tokens=1024)
print(tokenizer.decode(output[0]))

# VLLM ways
llm = LLM(path, tensor_parallel_size=1, quantization="awq")
sampling_params = SamplingParams(max_tokens=1024, n=4, temperature=0, use_beam_search=True)
output = llm.generate(prompt, sampling_params)[0].outputs[0].text
print(output)

**Comment by matrixssy** [comment]
I have tried to troubleshoot the following points:

1. The inference results of the original StarCoder model, vllm and hf, are different, but they are within an acceptable range.
2. After training, I confirmed that I did not modify the vocabulary or the size of the model word embedding.
3. The inference results of the trained model, vllm and hf, are vastly different. The inference results of vllm are unusable.
